### PR TITLE
Handle 500+ status codes when requesting a SoUD sync

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Fixed some assertions in tests
* Added test for handling ConnectionError in request
* Fixed issue where using `requests.exceptions.ConnectionError` made Python report that it was not derived from `BaseException`
* Handle 500+ status codes from the queue endpoint by initiating ttl back off

## Reviewer guidance
Tests cover the changes?

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/public/utils.py [hunk 1]
```diff
@@ -222,8 +223,9 @@ def request_soud_sync(server, user, queue_id=None, ttl=4):
             # got a place in the queue to sync with this server, so we can be
             # more sure that the server is actually available.
             response = requests.put(server_url, json=data, timeout=30)
-
-    except requests.exceptions.ConnectionError:
+        if response.status_code >= status.HTTP_500_INTERNAL_SERVER_ERROR:
+            raise ConnectionError()
```
Comment (jamalex) [inline kolibri/core/public/utils.py:227]: I'm assuming there's a reason we don't want to use `raise_for_status` here, yeah? Are there cases where we want 4xx response statuses to be swallowed?

**Review by jamalex — APPROVED** [review]
I don't know the scheduling code well enough to say anything definitive around how the tests work, but the code change itself seems wise (though I added a comment about `raise_for_status` as a possible alternative pattern, noting that it raises `requests.exceptions.HTTPError` instead, so we'd then need to catch that too).

File: kolibri/core/public/utils.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/public/utils.py:227]: Yeah - any 4xx error codes indicate that the server has received and processed the request, and is not interested, so we just stop trying.

**Review by jredrejo — COMMENTED** [review]
LGTM, thank you!

**Pull request merged**